bftengine/src/bftengine/ReplicaImp.cpp

```diff
@@ -293,23 +293,34 @@ void ReplicaImp::tryToSendPrePrepareMsg(bool batchingLogic) {
   PrePrepareMsg *pp =
       new PrePrepareMsg(config_.replicaId, curView, (primaryLastUsedSeqNum + 1), firstPath, primaryCombinedReqSize);
 
-  ClientRequestMsg *nextRequest = requestsQueueOfPrimary.front();
-  while (nextRequest != nullptr && nextRequest->size() <= pp->remainingSizeForRequests()) {
-    MDC_CID_PUT(GL, nextRequest->getCid());
-    if (clientsManager->noPendingAndRequestCanBecomePending(nextRequest->clientProxyId(),
-                                                            nextRequest->requestSeqNum())) {
-      pp->addRequest(nextRequest->body(), nextRequest->size());
-      clientsManager->addPendingRequest(nextRequest->clientProxyId(), nextRequest->requestSeqNum());
+  ClientRequestMsg *nextRequest = (!requestsQueueOfPrimary.empty() ? requestsQueueOfPrimary.front() : nullptr);
+
+  while (nextRequest != nullptr) {
+    if (nextRequest->size() <= pp->remainingSizeForRequests()) {
+      MDC_CID_PUT(GL, nextRequest->getCid());
+      if (clientsManager->noPendingAndRequestCanBecomePending(nextRequest->clientProxyId(),
+                                                              nextRequest->requestSeqNum())) {
+        pp->addRequest(nextRequest->body(), nextRequest->size());
+        clientsManager->addPendingRequest(nextRequest->clientProxyId(), nextRequest->requestSeqNum());
+      }
+      primaryCombinedReqSize -= nextRequest->size();
+    } else if (pp->numberOfRequests() <= 0) {  // case the first message is to big
+      LOG_ERROR(GL,
+                "Request sent by: " << nextRequest->senderId() << " Request size is: " << nextRequest->size()
```

Comment:
I think you should make this error message more explicit. It should say why the request is being dropped. It's not because it doesn't fit in the PrePrepare, it's because its larger than the maximum allowed size of a request.

Reply:
done.